todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModel.java

```diff
@@ -102,31 +85,24 @@ public Boolean call(Task task) {
                 return task.isActive();
             }
         }).count();
-        Subscription subscription = Observable
+
+        return Observable
                 .zip(completedTasks, activeTasks, new Func2<Integer, Integer, Pair<Integer, Integer>>() {
                     @Override
                     public Pair<Integer, Integer> call(Integer completed, Integer active) {
                         return Pair.create(active, completed);
                     }
                 })
-                .subscribeOn(mSchedulerProvider.computation())
-                .observeOn(mSchedulerProvider.ui())
-                .subscribe(new Action1<Pair<Integer, Integer>>() {
-                    @Override
-                    public void call(Pair<Integer, Integer> stats) {
-                        mStatisticsView.showStatistics(stats.first, stats.second);
-                    }
-                }, new Action1<Throwable>() {
-                    @Override
-                    public void call(Throwable throwable) {
-                        mStatisticsView.showLoadingStatisticsError();
-                    }
-                }, new Action0() {
+                .doOnCompleted(new Action0() {
```

Comment:
Setting progress to true might be made here like ``.doOnSubscribe(() -> mProgressIndicatorSubject.onNext(true))``